pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -16,24 +18,57 @@ import (
 func (a Adapter) createComponent() (err error) {
 	componentName := a.ComponentName
 
-	volumeLabels := utils.GetProjectVolumeLabels(componentName)
+	// Get or create the project source volume
+	var projectVolumeName string
+	projectVolumeLabels := utils.GetProjectVolumeLabels(componentName)
+	vols, err := a.Client.GetVolumesByLabel(projectVolumeLabels)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to retrieve source volume for component "+componentName)
+	}
+	if len(vols) == 0 {
+		// A source volume needs to be created
+		volume, err := a.Client.CreateVolume("", projectVolumeLabels)
```

Comment:
maybe now we can create a unique docker vol name for projects? Fine otherwise, like I said earlier as well.

Reply:
I'll use the `GenerateVolNameFromDevfileVol` to create a volume for the source

Reply:
Updated.